components/layout_2020/formatting_contexts.rs

```diff
@@ -54,6 +56,8 @@ pub(crate) struct ReplacedFormattingContext {
 pub(crate) enum NonReplacedFormattingContextContents {
     Flow(BlockFormattingContext),
     Flex(FlexContainer),
+    TaffyFlex(TaffyContainer),
```

Comment:
Instead of adding a new formatting context type here, please simply implement a new "taffy_layout" method for `NonReplacedFormattingContextContents::Flex` that's called conditionally.

Reply:
That isn't going to be very easy to do I'm afraid, as the `TaffyItemBox` type has extra fields compared to the main Flexbox implementation's item representation. It would however be trivial to combine the `TaffyFlex` and `Grid` variants if that was considered desirable (you'll notice they use the same inner type already). I could also combine the contexts by adding the new fields to the main flexbox impl's context (perhaps wrapped in `Option<Box<>>`) if you want?

A third option would be to keep separate contexts but abstract the item collection logic so that it is shared. My intuition on this is that it might make sense to do in future, but I am a little wary of entangling the implementations at this point.

FYI, I suspect if/when Servo gains a non-Taffy based Grid impl that will also be able to share a context with Flexbox, as both layout modes' treatment of items is identical.